site/app/entities/forum/Thread.php

```diff
@@ -48,12 +75,40 @@ class Thread {
     #[ORM\Column(type: Types::INTEGER)]
     protected int $status;
 
-    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE)]
-    protected DateTime $lock_thread_date;
+    public function getStatus(): int {
+        return $this->status;
+    }
+
+    public function setStatus(int $newStatus): void {
+        $this->status = $newStatus;
+    }
+
+    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE, nullable: true)]
+    protected ?DateTime $lock_thread_date;
+
+    public function getLockDate(): ?DateTime {
+        return $this->lock_thread_date;
+    }
+
+    public function isLocked(): bool {
+        return !is_null($this->lock_thread_date) && $this->lock_thread_date < date("Y-m-d H:i:S");
```

Comment:
Would it be better to use a DateTime object instead of comparing date strings here?

Also:
```suggestion
        return $this->lock_thread_date !== null && $this->lock_thread_date < date("Y-m-d H:i:S");
```